extension/githubappauthextension/extension.go

```diff
@@ -0,0 +1,41 @@
+package githubappauthextension // import "github.com/liatrio/liatrio-otel-collector/extension/githubappauthextension"
+
+import (
+	"net/http"
+
+	"github.com/bradleyfalzon/ghinstallation/v2"
+	"go.uber.org/zap"
+)
+
+// githubAppAuthenticator provides a simple struct to contain an http client
```

Comment:
- [ ] adjust this comment to more accurately reflect why we define this type to return the client and logger